Entities/Activity.cpp

```diff
@@ -0,0 +1,1136 @@
+		m_TeamIcons[Activity::TEAM_4] = *dynamic_cast<const Icon *>(g_PresetMan.GetEntityPreset("Icon", "Team 4 Default"));
+
+		return 0;
+	}
+
+/////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////
+
+	int Activity::Create(const Activity &reference) {
+		Entity::Create(reference);
+
+		m_Description = reference.m_Description;
+		m_InCampaignStage = reference.m_InCampaignStage;
+		m_MaxPlayerSupport = reference.m_MaxPlayerSupport;
+		m_MinTeamsRequired = reference.m_MinTeamsRequired;
+		m_ActivityState = reference.m_ActivityState;
+		m_Paused = reference.m_Paused;
+		m_TeamCount = reference.m_TeamCount;
+		m_PlayerCount = reference.m_PlayerCount;
+		m_Difficulty = reference.m_Difficulty;
+		m_CraftOrbitAtTheEdge = reference.m_CraftOrbitAtTheEdge;
+
+		for (int player = 0; player < MAXPLAYERCOUNT; ++player) {
+			m_Team[player] = reference.m_Team[player];
+			m_FundsContribution[player] = reference.m_FundsContribution[player];
+			m_TeamFundsShare[player] = reference.m_TeamFundsShare[player];
+			m_IsActive[player] = reference.m_IsActive[player];
+			m_IsHuman[player] = reference.m_IsHuman[player];
+			m_PlayerScreen[player] = reference.m_PlayerScreen[player];
+			m_Brain[player] = reference.m_Brain[player];
+			m_HadBrain[player] = reference.m_HadBrain[player];
+			m_BrainEvacuated[player] = reference.m_BrainEvacuated[player];
+			m_ViewState[player] = reference.m_ViewState[player];
+			m_ControlledActor[player] = reference.m_ControlledActor[player];
+			m_PlayerController[player] = reference.m_PlayerController[player];
+		}
+
+		for (int team = 0; team < MAXTEAMCOUNT; ++team) {
+			m_TeamActive[team] = reference.m_TeamActive[team];
+			m_TeamFunds[team] = reference.m_TeamFunds[team];
+			m_FundsChanged[team] = reference.m_FundsChanged[team];
+			m_TeamNames[team] = reference.m_TeamNames[team];
+			if (reference.m_TeamIcons[team].GetFrameCount() > 0) { m_TeamIcons[team] = reference.m_TeamIcons[team]; }		
+			m_TeamDeaths[team] = reference.m_TeamDeaths[team];
+			m_TeamAISkillLevels[team] = reference.m_TeamAISkillLevels[team];
+		}
+
+		return 0;
+	}
+
+/////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////
+
+	int Activity::ReadProperty(std::string propName, Reader &reader) {
+		if (propName == "Description") {
+			reader >> m_Description;
+		} else if (propName == "SceneName") {
+			reader >> m_SceneName;
+		} else if (propName == "InCampaignStage") {
+			reader >> m_InCampaignStage;
+		} else if (propName == "MaxPlayerSupport") {
+			reader >> m_MaxPlayerSupport;
+		} else if (propName == "MinTeamsRequired") {
+			reader >> m_MinTeamsRequired;
+
+		// Gets tallied as team assignments are read in below, so doesn't really matter here, but should still be handled
+		} else if (propName == "TeamCount") {
+			reader.ReadPropValue(); //reader >> m_TeamCount;
```

Comment:
Yeah i've actually been thinking about removing the `TeamCount` and `PlayerCount` properties since they both do nothing.
Gonna break a few more mods, but oh well. ¯\\\_(ツ)_/¯

Reply:
Blow em up! Seriously, I'm all for removing them.